### DISCUSSION TITLE
Far Field Plots - Contours?

### DISCUSSION THREAD

**Discussion by zekishaw** [NanoComp/meep#2639] [General]
![image](https://github.com/NanoComp/meep/assets/70750128/e3b8d850-382d-4540-a1e2-8c59b36ddf15)

In Lumerical, you can plot these sort of far field plots with colourmaps and the like. I was wandering if MEEP had the same functionality? 

I'm a PhD student but the group I work within (or one of) does optical characterisation and nanofab, and the hope is that some of my designs will one day be fabricatable - The far field plots are useful for seeing how much light can be directed to collection optics and what collection angle, and I really don't like using Lumerical so was hoping that this a function that MEEP supports? I would imagine yes, but also it is not shown in the tutorials, unless I have missed it in which case apologies, a link would be appreciated!!

**Comment by zekishaw** [comment]
This is a generic one taken literally from their website but shows what I want to do. Secondarily, is there a version of teh docs that only contains the python interface? And tutorials for the C++? I am hoping for some speedup and hoping the C++ does this but please corect me if wrong!

**Reply by smartalecH** [reply]
The core timestepping is in C++. The python interface calls down to the core library directly. Using the C++ interface won't bring any additional speedup.

**Reply by zekishaw** [reply]
Ah I see! Thank you - I thought there may be some slowdown due to communication between the C++ routines and core python. Glad to hear this!

**Comment by smartalecH** [comment]
This is just a simple postprocessing visualization step. You can take any of the Near2Far tutorials to collect the powers in the far field for a large collection of angle pairs. Then you simply plot them how you wish. 

To get exactly this plot, there are plenty of examples found on google, like this one: https://stackoverflow.com/questions/54209640/imshow-in-polar-coordinates

**Reply by zekishaw** [reply]
This is brilliant thank you!

**Comment by oskooi** [comment]
For an example of computing the radiation pattern in cylindrical coordinates using Matplotlib, see the functions `plot_radiation_pattern_polar` and `plot_radiation_pattern_3d` and the corresponding figure in #2629.
